src/pool/srv_pool.c

```diff
@@ -368,6 +368,15 @@ pool_prop_write(struct rdb_tx *tx, const rdb_path_t *kvs, daos_prop_t *prop)
 		entry = &prop->dpp_entries[i];
 		switch (entry->dpe_type) {
 		case DAOS_PROP_PO_LABEL:
+			if (entry->dpe_str == NULL ||
+			    strlen(entry->dpe_str) == 0) {
+				entry = daos_prop_entry_get(&pool_prop_default,
+							    entry->dpe_type);
+				if (entry == NULL) {
+					D_ERROR("no default label property");
+					return -DER_INVAL;
+				}
```

Comment:
I'd be tempted to D_ASSERT on this case. Only way we'd end up with no default is if developers removed it. In either case I don't think -DER_INVAL is the right code -- something surprising has gone wrong if we end up here.

Reply:
Agreed.